reduct/boolean-reduct/cut-unnecessary-or.metta

```diff
@@ -0,0 +1,24 @@
+;a function which implements the Cut-Unnecessary-OR Transformation
+;POA: Any OR node
+;If the POA has only 1 child, orCut returns the child's guardSet and children to the POA's parent
+;EX: (OR (AND A (OR C D))) -> ( (A) (A (OR C D)) )
+(= (orCut $exp)  
+    (let*
+        (
+            ($op (car-atom $exp))
+            ($children (cdr-atom $exp))
+            ($child (car-atom $children)) 
+            ($restChildren (cdr-atom $children))  
+            ($grandChildren (if (== (get-metatype $child) Symbol) () (cdr-atom $child)))  
+        )
+        (case $op
+        (
+            (AND $exp)
+            (NOT $exp)
```

Comment:
@Liya-F  why did you specify both AND and NOT on your case statements?

Reply:
> @Liya-F why did you specify both AND and NOT on your case statements?

Because the POA is an OR node, if the expression starts with AND or NOT, it is returned as it is. Should I only specify the condition for an OR node and put the rest inside $else?

Reply:
@Liya-F it is fine It seemed to me as a redundant condition